### DISCUSSION TITLE
Need to show timestamp with time zone field in a different time zone in the report

### DISCUSSION THREAD

**Discussion by gogitub** [apache/superset#14281] [General]
Hi,
We have many timestamp with time zone fields in Oracle database which is the source for our Superset (0.36 version) reports.
We need to show the data in EST or PST or some other time zones based on customer need.

Ex:
Data in Oracle database table is like '2021-01-04 21:48:56.66 +05:30'
If we want to display in EST, report should show it like : 2021-01-04T11:18:56.660000
If we want to display in PST, report should show it like : 2021-01-04T08:18:56.660000

So, is there any configuration parameter that we can modify to display the data in desired time zone (like EST/PST or some other)?

Please advise. Thanks in advance.

**Comment by zhaoyongjie** [comment]
@gogitub. Is it possible to add derived dimensions from the original date field ?

**Comment by gogitub** [comment]
@zhaoyongjie, Thank you for your interest in answering my question.
There are a lot of such fields and many time zones are available in the data. we are not preferring any derived columns because we already have lot many reports/dashboards built with the existing columns.
Also, we need to show the data displayed in time zone per user profile or preference if there is a way.

**Comment by gogitub** [comment]
It is observed that Superset is automatically converting the timestamp fields to EST time while displaying in the SQL Lab or Chart.
Anyone knows where exactly we can change this setting to show it in different time zone or customize it to work using a config parameter?

**Comment by eugeniamz** [comment]
Have you tried the HOURS OFFSET from Dataset / Setting?
![image](https://user-images.githubusercontent.com/58375897/104192915-7af28200-53ed-11eb-8bde-954bd045b7ab.png)

I am not sure if this is your solution, would like to know your feedback.

**Comment by gogitub** [comment]
Not this datasource specific setting. 
Currently, any TIMESTAMP with TIME ZONE field from source DB (Oracle) is being displayed in EST time zone in our application (I think it is converting to the system time zone where it is installed)
